### PR TITLE
Div component

### PR DESCRIPTION
**Pull request by MonPote**
**Component**:

**Description**:

**Design**:

**Breaking Changes**:

- [] Breaking Changes

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: src/lib/components/div/Div.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  return result;
+};
+
+const m = createSpacingFn('m', ['margin']);
+const mt = createSpacingFn('mt', ['margin-top']);
+const mr = createSpacingFn('mr', ['margin-right']);
+const mb = createSpacingFn('mb', ['margin-bottom']);
+const ml = createSpacingFn('ml', ['margin-left']);
+const mx = createSpacingFn('mx', ['margin-top', 'margin-bottom']);
+const my = createSpacingFn('my', ['margin-left', 'margin-right']);
```
Comment (JBWatenbergScality) [inline src/lib/components/div/Div.js]: I think x and y axis are reversed here. In the common spirit x should stand for the horizontal axis I believe. (Tailwind also implemented the x axis against left/right properties : https://tailwindcss.com/docs/padding#add-horizontal-padding)

**Review by JBWatenbergScality — COMMENTED** [review]
Additionally to the raw component we might want to document/show some example of the composability of this approach when it comes to use those spacing utils with other components

File: src/lib/components/div/Div.js (hunk 1 above)
Comment (MonPote) [reply src/lib/components/div/Div.js]: Indeed that was a mistake

File: stories/div.stories.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      </Div>
+      <Div as={BoldP} my={14} p={14}>
+        Bold {'"</p>"'} with my 14 and p 14
+      </Div>
+      <Div
+        as={DisplayTextComp}
+        text={'Div + React component + nested styled component'}
+        m={12}
+        p={14}
+      ></Div>
```
Comment (JBWatenbergScality) [inline stories/div.stories.js]: This is great, I'm wondering how does it affect the typing ? Do we get props autocompletion/validation for both Div and  DisplayTextComp component ?

**Review by JBWatenbergScality — APPROVED** [review]

**Pull request merged**